src/api/app/models/issue.rb

```diff
@@ -10,6 +10,21 @@ class NotFoundError < APIError
   belongs_to :issue_tracker
   belongs_to :owner, class_name: 'User'
 
+  validates :name, on: :create, format: {
+    with: ->(issue) { Regexp.new(issue.issue_tracker.regex) },
+    message: ->(issue, data) { "with value \'#{data[:value]}\' does not match defined regex #{issue.issue_tracker.regex}" }
+  }
+
+  # rubocop:disable Lint/UselessAssignment, Lint/UnneededCopDisableDirective, Style/RedundantSelf
+  before_validation do
+    self.name = issue_tracker.show_label_for(self.name)
+  end
+
+  after_validation do
+    self.name = name[Regexp.new(self.issue_tracker.regex), 1]
+  end
```

Comment:
The before / after validation changes look like a workaround. Instead, please fetch the label within the validation. This might mean that you need a custom validation method, e.g. something like this (untested):

```ruby
def name_validation
  return if issue_tracker.show_label_for(self.name) =~ Regexp.new(issue.issue_tracker.regex)
  errors.add(:name, $some_message)
end
```

That's what we also do in some of the other models. If you want to go one step further you could custom validator class as described in the article you posted. But I think the method above would already be good enough.

Reply:
I hope, this version meets your requirement.

Reply:
@bgeuken Do I need to make more changes to this PR?